src/test/java/com/exasol/dbbuilder/dialects/mysql/MySQLDatabaseObjectCreationIT.java

```diff
@@ -0,0 +1,158 @@
+        assertAll(() -> assertUserHasGlobalPrivilege(user.getName(), CREATE_USER, "Create_user_priv"),
+                () -> assertUserHasGlobalPrivilege(user.getName(), CREATE_ROLE, "Create_role_priv"));
+    }
+
+    private void assertUserHasGlobalPrivilege(final String username, final GlobalPrivilege expectedPrivilege,
+            final String columnName) throws AssertionError {
+        final String sql = "SELECT `" + columnName + "` FROM mysql.user WHERE User = '" + username + "'";
+        assertObjectExistsInDatabase(sql, "Y");
+    }
+
+    @Test
+    void testGrantSchemaPrivilegeToUser() throws InterruptedException {
+        final Schema schema = this.factory.createSchema("OBJPRIVSCHEMA");
+        final User user = this.factory.createUser("OBJPRIVUSER").grant(schema, SELECT, DELETE);
+        assertAll(() -> assertUserHasSchemaPrivilege(user.getName(), schema.getName(), SELECT, "Select_priv"),
+                () -> assertUserHasSchemaPrivilege(user.getName(), schema.getName(), DELETE, "Delete_priv"));
+    }
+
+    private void assertUserHasSchemaPrivilege(final String username, final String objectName,
+            final ObjectPrivilege expectedPrivilege, final String columnName) throws AssertionError {
+        final String sql = "SELECT `" + columnName + "` FROM mysql.db WHERE User = '" + username + "' AND Db = '"
+                + objectName + "'";
+        assertObjectExistsInDatabase(sql, "Y");
+    }
+
+    @Test
+    void testGrantTablePrivilegeToUser() throws InterruptedException {
+        final Schema schema = this.factory.createSchema("TABPRIVSCHEMA");
+        final Table table = schema.createTable("TABPRIVTABLE", "COL1", "DATE", "COL2", "INT");
+        final User user = this.factory.createUser("TABPRIVUSER").grant(table, SELECT, DELETE);
+        assertAll(() -> assertUserHasTablePrivilege(user.getName(), table.getName(), "Select"),
+                () -> assertUserHasTablePrivilege(user.getName(), table.getName(), "Delete"));
+    }
+
+    private void assertUserHasTablePrivilege(final String username, final String objectName,
+            final String expectedPrivilege) throws AssertionError {
+        try (final PreparedStatement statement = this.adminConnection
+                .prepareStatement("SELECT `Table_priv` FROM mysql.tables_priv WHERE User = '" + username
+                        + "' AND Table_name = '" + objectName + "'")) {
+            final ResultSet result = statement.executeQuery();
+            assertAll(() -> assertThat(result.next(), equalTo(true)),
+                    () -> assertThat(result.getString(1), containsString(expectedPrivilege)));
+        } catch (final SQLException exception) {
+            throw new AssertionError(
+                    "Unable to determine if user " + username + " has global privilege " + expectedPrivilege + ".",
+                    exception);
+        }
+    }
+
+    @Test
+    void testInsertIntoTable() {
```

Comment:
Use the new `ResultSetStructureMatcher` to reduce this test.

Reply:
Somehow done...

Reply:
When comparing numbers, `fuzzyMatch()` makes you life easier.

Reply:
Didn't help in my case

Reply:
Issue ticket exists.